src/decoder/zlib.rs

```diff
@@ -98,17 +112,34 @@ impl ZlibStream {
             in_pos: 0,
             out_buffer: vec![0; 2 * CHUNCK_BUFFER_SIZE],
             out_pos: 0,
+            ignore_adler32: true,
         }
     }
 
     pub(crate) fn reset(&mut self) {
         self.started = false;
         self.in_buffer.clear();
+        self.in_pos = 0;
```

Comment:
This seemed like an oversight but I'm not sure since nothing fails with this reset or not. I'm not clear on what the documentation means:
```rust
    /// The logical start of the `in_buffer`.
    in_pos: usize
```

L172 is as follows, should this be `<=`?
```rust
        if self.in_buffer.len() == self.in_pos {
            self.in_buffer.clear();
            self.in_pos = 0;
        }
```

Reply:
Huh. That seems plausible but I can't really say either. I don't think the reset code path is tested as exhaustive as it could be.